rabbitmq/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/RabbitMQSink.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2013-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.collectors.scalastream
+package sinks
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+import cats.syntax.either._
+
+import com.rabbitmq.client.Channel
+
+import com.snowplowanalytics.snowplow.collectors.scalastream.model.RabbitMQBackoffPolicyConfig
+
+class RabbitMQSink(
+  channel: Channel,
+  exchangeName: String,
+  backoffPolicy: RabbitMQBackoffPolicyConfig
+) extends Sink {
+
+  implicit val ec = ExecutionContext.global
+
+  override val MaxBytes = 128000000
+
+  override def storeRawEvents(events: List[Array[Byte]], key: String): Unit =
+    if (events.nonEmpty) {
+      log.info(
+        s"Sending ${events.size} Thrift records to exchange $exchangeName"
+      )
+      Future.sequence(events.map(e => sendOneEvent(e))).onComplete {
+        case Success(_) =>
+          log.debug(
+            s"${events.size} events successfully sent to exchange $exchangeName"
+          )
+        // We should never reach this as the writing of each individual event is retried forever
+        case Failure(e) =>
+          throw new RuntimeException(s"Error happened during the sending of ${events.size} events: ${e.getMessage}")
+      }
+    }
+
+  private def sendOneEvent(bytes: Array[Byte], currentBackOff: Option[FiniteDuration] = None): Future[Unit] =
+    Future {
+      if (currentBackOff.isDefined) Thread.sleep(currentBackOff.get.toMillis)
+      channel.basicPublish(exchangeName, "", null, bytes)
```

Comment:
Would it be worth to be able to allow setting routing keys? This allows you to build pretty impressive topologies. 
This one in particular would allow us to deploy canary-builds. Setting routing key to `latest` on 10% of collector instances when the rest of the instances are `stable` allow us to have a deployment-check for consumer for the `unstable`. If succeeds we can roll out 100%.
This would also allow having an interesting topology where parts of the messages are delivered to both enrich and another destination. 
Anyway, I feel like it might be worth it to expose it as a config parameter.

Reply:
First it was exposed and then I removed it because when we were testing Josh said that we would not use it but I agree that it doesn't hurt to expose it anyway.

Reply:
I've spent some quality time using and abusing rabbit before and the ability to use various topologies is what is really exciting in this PR. I think having that option might come in handy to some OSS users who'd like to use their existing rabbit setup. Especially as this partially implements multiple purpose pipelines people have been asking before. 
I think part of the announcement, apart from the cloud-agnosticism, could be "all the cool things you could do with the new transport". And topologies are really interesting topic.

Reply:
I've spent some quality time using and abusing rabbit in my previous life. And part of my excitement for this new transport layer is being able to set up those topologies. Apart from the cloud-agnosticism, it feels like a "feature" we should be able to use as a part of the announcement. "All the cool topologies you could build with the new collector/enrich".